src/index.ts

```diff
@@ -508,15 +520,64 @@ export class ServerlessClamscan extends Construct {
     }
   }
 
+  /**
+   * @returns ArnPrincipal the ARN of the assumed role principal for the scan function
+   */
+  get scanAssumedPrincipal(): ArnPrincipal {
+    if (this._scanFunction.role) {
+      const stack = Stack.of(this);
+      const scan_assumed_role = `arn:${stack.partition}:sts::${stack.account}:assumed-role/${this._scanFunction.role.roleName}/${this._scanFunction.functionName}`;
+      return new ArnPrincipal(scan_assumed_role);
+    } else {
+      throw new Error('The scan function role is undefined');
+    }
+  }
+
+
+  /**
+   * Returns the statement that should be added to the bucket policy
+     in order to prevent objects to be accessed when they are not clean
+     or there have been scanning errors: this policy should be added
+     manually if external buckets are passed to addSourceBucket()
+   * @param bucket The bucket which you need to protect with the policy
+   * @returns PolicyStatement the policy statement if available
+   */
+  getPolicyStatementForBucket(bucket: IBucket): PolicyStatement {
+    if (this._scanFunction.role) {
+      const scan_assumed_principal = this.scanAssumedPrincipal;
+      return new PolicyStatement({
+        effect: Effect.DENY,
+        actions: ['s3:GetObject'],
+        resources: [bucket.arnForObjects('*')],
+        notPrincipals: [this._scanFunction.role, scan_assumed_principal],
+        conditions: {
+          StringEquals: {
+            's3:ExistingObjectTag/scan-status': [
+              'IN PROGRESS',
+              'INFECTED',
+              'ERROR',
+            ],
+          },
+        },
+      });
+    } else {
+      throw new Error("Can't generate a valid S3 bucket policy, the scan function role is undefined");
+    }
+  }
+
   /**
    * Sets the specified S3 Bucket as a s3:ObjectCreate* for the ClamAV function.
      Grants the ClamAV function permissions to get and tag objects.
      Adds a bucket policy to disallow GetObject operations on files that are tagged 'IN PROGRESS', 'INFECTED', or 'ERROR'.
    * @param bucket The bucket to add the scanning bucket policy and s3:ObjectCreate* trigger to.
    */
-  addSourceBucket(bucket: Bucket) {
+  addSourceBucket(bucket: IBucket) {
+    if (!(bucket instanceof Bucket) && !this.useExternalBuckets) {
+      throw new Error('acceptResponsibilityForUsingExistingBucket must be set when adding an external bucket');
```

Comment:
Let's be more descriptive here.
```suggestion
      throw new Error("acceptResponsibilityForUsingImportedBucket must be set when adding an imported bucket. When using imported buckets the user is responsible for adding the required policy statement to the bucket policy: `getPolicyStatementForBucket()` can be used to retrieve the policy statement required by the solution");
```